apps/typegpu-docs/src/content/docs/fundamentals/resolve.mdx

```diff
@@ -0,0 +1,124 @@
+---
+title: Resolve
+description: Resolve API can be used to extend shader code with WGSL definitions of TypeGPU-defined objects.
+---
+
+Defining shader schemas and objects in JS/TS has lots of benefits, but for a while it meant introducing significant duplication between host and GPU code.
+In addition to that it was crucial to keep both sides in sync manually when making any changes.
+To mitigate this problem TypeGPU offers the **`tgpu.resolve`** API, which allows there to be just one source of truth,
+ensuring that the shader code is always up-to-date with the latest changes to the data schemas, bind group layouts and such. 
+It takes WGSL code and all the external objects it uses, but doesn't declare, and returns the complete necessary WGSL code, ready to be used as part of a *GPUShaderModule*.
+
+:::note
+`tgpu.resolve` is essentially a dedicated TypeGPU linker.
+:::
+
+Here's an example:
+```ts
+import tgpu from 'typegpu';
+import * as d from 'typegpu/data';
+
+const LightSource = d.struct({
+  ambientColor: d.vec3f,
+  intensity: d.f32,
+}).$name('Source');
```

Comment:
Lets add more context, just so that devs know they do not have to name every structs.
```suggestion
}).$name('Source');
// ^ Giving the struct an explicit name
```